api/main.py

```
@@ -54,87 +87,146 @@ async def root():
+
+        # Firestore operations
+        get_or_create_user(current_user["uid"], current_user.get("email"))
+        save_prediction(
+            uid=current_user["uid"],
+            assessment=assessment,
+            feature_values=feature_dict,
+            transaction_count=len(df),
+            file_name=file.filename
+        )
+
         return assessment
```

Ruff (0.14.4)

Consider moving this statement to an `else` block

(TRY300)